python/eups/utils.py

```diff
@@ -151,12 +151,12 @@ def setupEnvNameFor(productName):
     return the name of the environment variable that provides the 
     setup information for a product.  This is of the form "setupEnvPrefix() + prod".
     """
-    name = setupEnvPrefix() + productName
+    name = setupEnvPrefix() + productName.upper()
 
     if os.environ.has_key(name):
         return name                 # exact match
 
-    envNames = filter(lambda k: re.search(r"^%s$" % name, k, re.IGNORECASE), os.environ.keys())
```

Comment:
Is this part really necessary at all?  Are we ever using lower/mixed-case env vars?  If so, what is special about the first one found that makes it correct?